### PR TITLE
Snooze VPN while calling internal festure

### PR DESCRIPTION
**Pull request by aitorvs**
Task/Issue URL: https://app.asana.com/0/42792087274227/1205827620980592/f

### Description
Add an internal setting to disable VPN while calling

### Steps to test this PR

_Test VPN disable while calling_
- [x] install from this branch
- [x] enable NetP (and optionally AppTP)
- [x] go to settings -> (scroll down) NetP dev settings and toggle "Disable VPN while calling" setting
- [x] verify PHONE permissions are handled properly
- [x] place a call (ring is enough)
- [x] verify VPN is disabled
- [x] end the call
- [x] verify VPN is re-enabled

### CODE REVIEW

**Comment by aitorvs** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #3779** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/3779" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/duckduckgo/Android/3779?utm_source=stack-comment).

File: app-tracking-protection/vpn-api/src/main/java/com/duckduckgo/mobile/android/vpn/Vpn.kt [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.mobile.android.vpn
+
+interface Vpn {
```
Comment (aitorvs) [inline app-tracking-protection/vpn-api/src/main/java/com/duckduckgo/mobile/android/vpn/Vpn.kt:19]: This is an API that I found missing. Currently we can enable/disable individual VPN features (AppTP/NetP) but we can't disable/enable the entire VPN without going directly through the vpn service.

File: common/common-utils/build.gradle [hunk 2]
```diff
@@ -45,6 +45,7 @@ dependencies {
 
     implementation AndroidX.work.runtimeKtx
     implementation AndroidX.lifecycle.viewModelKtx
+    implementation AndroidX.appCompat
```
Comment (aitorvs) [inline common/common-utils/build.gradle:48]: Needed for the AppCompatActivity

File: network-protection/network-protection-internal/src/main/AndroidManifest.xml [hunk 3]
```diff
@@ -6,6 +6,7 @@
     <uses-permission
         android:name="android.permission.QUERY_ALL_PACKAGES"
         tools:ignore="QueryAllPackagesPermission" />
+    <uses-permission android:name="android.permission.READ_PHONE_STATE" />
```
Comment (aitorvs) [inline network-protection/network-protection-internal/src/main/AndroidManifest.xml:9]: note: only added in internal builds, won't affect prod

File: network-protection/network-protection-internal/src/main/java/com/duckduckgo/networkprotection/internal/feature/snooze/VpnCallStateReceiver.kt [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+
+    private val telephonyManager by lazy {
+        context.getSystemService(Context.TELEPHONY_SERVICE) as TelephonyManager?
+    }
+    private val _listener: PhoneStateListener =
+        object : PhoneStateListener() {
+            @Deprecated("Deprecated in Java")
+            override fun onCallStateChanged(state: Int, phoneNumber: String?) {
+                appCoroutineScope.launch(dispatcherProvider.io()) {
+                    logcat { "Call state: $state" }
```
Comment (karlenDimla) [inline network-protection/network-protection-internal/src/main/java/com/duckduckgo/networkprotection/internal/feature/snooze/VpnCallStateReceiver.kt:80]: Note that regardless the state of the VPN, this is being called as long as the internal setting is enabled. I think we should only call this if the vpnn has been enabled prior.

I see that when VPN is disabled every after a call:

- VPN log: onCreate, creating the APP_TP_VPN network stack
- VPN log: VPN network APP_TP_VPN created
- VPN log: onStartCommand: ACTION_START_VPN
- VPN log: onDestroy

Comment (aitorvs) [reply network-protection/network-protection-internal/src/main/java/com/duckduckgo/networkprotection/internal/feature/snooze/VpnCallStateReceiver.kt:80]: Yeah, the difficulty I found doing that eg. in vpn callbacks is that the receiver needs to outlive the VPN service (needs to re-enable the VPN once call is done).
Lmk if you have ideas, open to them

**Review by karlenDimla — APPROVED** [review]

**Pull request merged**